src/pages/api/admin/workspaces/session.ts

```diff
@@ -1,25 +1,34 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
 import { AdminWorkspaceCRData } from '../../../../../types/AdminWorkspaceCRData';
-import appConfig from '../../../../../configs/app_config';
-import { TheiaServiceClient } from '../../../../../utils/theiaservice/theiaservice_client';
+// import appConfig from '../../../../../configs/app_config';
+import hostConfig from '../../../../../configs/ws_config';
+// import { TheiaServiceClient } from '../../../../../utils/theiaservice/theiaservice_client';
+import { KubernetesClient } from '../../../../../utils/k8s/k8s_client';
 
-// process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  const theiaService = new TheiaServiceClient(req.headers['x-access-token']);
+  // const theiaService = new TheiaServiceClient(req.headers['x-access-token']);
+  const k8s = new KubernetesClient();
   if (req.method === 'POST') {
     try {
       await Promise.all(
         req.body.toBeStartedSessions.map(async (workspace: AdminWorkspaceCRData) => {
-          await theiaService.createSessionWithExistingWorkspace(
-            appConfig.appId,
-            workspace.user,
-            workspace.name,
-            workspace.appDefinition
-          );
+          let sessionName = workspace.name + '-session'
+          let sessionUrl = hostConfig.hostUrl + '/' + workspace.uid + '/'
+          await k8s.createSession(sessionName, workspace.name, workspace.appDefinition, sessionUrl, workspace.user);
+          // Theia Cloud Service is checking some keycloak configurations, our roles are maybe not recognized by your service api. 
+          // After arranging that you can change it to theia api back again.
+          // await theiaService.createSessionWithExistingWorkspace(
+          //   appConfig.appId,
+          //   workspace.user,
+          //   workspace.name,
+          //   workspace.appDefinition
+          // );
```

Comment:
Is this about the bug I initially mentioned? If so, It isn't only related to restarting sessions, but to all theia.cloud API. Thus, it doesn't make sense to switch only this back to K8 API. Also, our methods to create/restart/edit sessions/workspaces using K8 API were mostly buggy, and we should stick to theai.cloud API. As I mentioned before, the workaround for this bug right now is to first login through user. Then you can use all Admin API. You only need to do this the first time you start your cluster.

Reply:
changed back